src/components/offers-list/offers-list.jsx

```diff
@@ -0,0 +1,52 @@
+import React, {PureComponent} from "react";
+import PropTypes from "prop-types";
+import OfferCard from "../offer-card/offer-card.jsx";
+
+class OffersList extends PureComponent {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      activeCardId: 0,
+    };
+
+    this._handleCardHover = this._handleCardHover.bind(this);
+    this._handleCardTitleClick = this._handleCardTitleClick.bind(this);
+  }
+
+  render() {
+    const {offers} = this.props;
+
+    return offers.map((offer) => {
+      const {id} = offer;
+      return <OfferCard
+        key={id}
+        offer={offer}
+        onCardHover={this._handleCardHover}
+        onTitleClick={this._handleCardTitleClick}
+      />;
+    });
+  }
```

Comment:
```suggestion
  render() {
    return this.props.offers.map((offer) => (<OfferCard
         key={offer.id}
         offer={offer}
         onCardHover={this._handleCardHover}
         onTitleClick={this._handleCardTitleClick}
       />)
    );
  }
```

Не усложняй. Советую выносить часть обьекта  в переменную только тогда, когда потом используешь ее более 2 раз.